### PR TITLE
Student, StudentList, University and UniversityRepository for DG

### PR DESCRIPTION
**Pull request by paulktham**
Added the necessary explanations and diagram for a simple developer's guide for these classes

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -100,14 +100,30 @@ The boolean return value of `parseInput()` indicates whether the user has chosen
...
 
+#### Student
+
+The `Student` class has a composition relationship with class StudentList. Its purpose is to store key information on the different students that have applied for the Student Exchange Program. Such information include their GPA and university preferences, which helps us allocate them to the various universities fairly, and also other information which helps the app track their allocation status.
+
+#### StudentList
+
+The ```StudentList``` is a fundamental component which is initiated as soon as FindOurSEP is initialised. Its purpose is to hold the necessary information of the different students that are applying for SEP. By having the list of students we are able to fairly allocate universities to the different students by comparing them to their cohort.
+
+The sequence below illustrates the interactions between ```StudentList``` and ```Student``` when and addCommand is called with the appropriate inputs.
```
Comment (ehz0ah) [inline docs/DeveloperGuide.md:111]: Do you mean "an addCommand is called..." instead?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -100,14 +100,30 @@ The boolean return value of `parseInput()` indicates whether the user has chosen
...
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+This was designed for admins handling the allocation of Student Exchange Program (SEP) locations for Computer Engineering (CEG) students at NUS.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+The app allows administrators to efficiently manage the allocation process using automated workflows and data-driven decision-making. Giving the administrators greater convenience when allocating students for their SEP.
```
Comment (ehz0ah) [inline docs/DeveloperGuide.md:126]: Nice!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -100,14 +100,30 @@ The boolean return value of `parseInput()` indicates whether the user has chosen
...
+
+The `Student` class has a composition relationship with class StudentList. Its purpose is to store key information on the different students that have applied for the Student Exchange Program. Such information include their GPA and university preferences, which helps us allocate them to the various universities fairly, and also other information which helps the app track their allocation status.
+
+#### StudentList
+
+The ```StudentList``` is a fundamental component which is initiated as soon as FindOurSEP is initialised. Its purpose is to hold the necessary information of the different students that are applying for SEP. By having the list of students we are able to fairly allocate universities to the different students by comparing them to their cohort.
+
+The sequence below illustrates the interactions between ```StudentList``` and ```Student``` when and addCommand is called with the appropriate inputs.
+
+![StudentListSequence](UML_Diagrams/StudentList.drawio.svg)
```
Comment (ehz0ah) [inline docs/DeveloperGuide.md:113]: - The tail of the arrow representing the constructor call of the Student class by the StudentList class can be more aligned with the activation bar
- Termination of the Student object should be somewhere along its lifeline

Comment (ehz0ah) [inline docs/DeveloperGuide.md:113]: Do you think it's better to show the interactions between the StudentList class and the Student class in the addStudent() method? Maybe through the use of a loop frame?

File: docs/Untitled Diagram.drawio [hunk 4]
```diff
@@ -0,0 +1,126 @@
+<mxfile host="app.diagrams.net" agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36" version="24.8.0">
```
Comment (ehz0ah) [inline docs/Untitled Diagram.drawio:1]: Is this file related to any diagram?

**Review by ehz0ah — CHANGES_REQUESTED** [review]
Good acknowledgement of the composition relationship between the Student and StudentList class! Only a few nits to fix :)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (paulktham) [reply docs/DeveloperGuide.md:113]: LOL shit

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (isaacsaw25) [reply docs/DeveloperGuide.md:115]: I don't think there is a loop frame since it is only adding 1 student

Comment (ehz0ah) [reply docs/DeveloperGuide.md:115]: There's actually a for loop used to check if the Student already exist within StudentList. Feel free to disregard my comments if this is too trivial :)

File: docs/UML_Diagrams/StudentList.drawio.svg
Comment (isaacsaw25) [inline docs/UML_Diagrams/StudentList.drawio.svg:1]: The loop condition should be enclosed in square brackets, and I think "checked" would be a more apt word. Perhaps consider this loop text: `[until all students are checked]`

**Review by isaacsaw25 — CHANGES_REQUESTED** [review]
Small fix needed 😅

**Review by isaacsaw25 — APPROVED** [review]
LGTM! 😃

**Review by ehz0ah — APPROVED** [review]
LGTM!
